test/Common/Test.hs

```diff
@@ -630,6 +630,28 @@ testSelectWhere run = do
                return p
         liftIO $ ret `shouldBe` [ p3e ]
 
+    it "works for a simple example with between and [uses just . val]" $
+      run $ do
+        p1e  <- insert' p1
+        _    <- insert' p2
+        _    <- insert' p3
+        ret  <- select $
+          from $ \p -> do
+            where_ ((p ^. PersonAge) `between` (just $ val 20, just $ val 40))
+            return p
+        liftIO $ ret `shouldBe` [ p1e ]
+    it "works for a SqlExpr value with between" $
+      run $ do
+        _ <- insert' p1 >> insert' p2 >> insert' p3
+        ret <-
+          select $
+          from $ \p -> do
+          where_ $
+            just (p ^. PersonFavNum)
+              `between`
+                (p ^. PersonAge, p ^.  PersonWeight)
+        liftIO $ ret `shouldBe` []
```

Comment:
Let's write some tests for composite keys, too. There's no reason we can't support them.